### PR TITLE
[TEST] Run tsdb tests with both base and trial licenses

### PR DESCRIPTION
**Pull request by kkrik-es**
Base license uses stored source, trial switches to synthetic source.

### CODE REVIEW

File: modules/data-streams/src/yamlRestTest/java/org/elasticsearch/datastreams/DataStreamsClientYamlTestSuiteIT.java [hunk 1]
```diff
@@ -50,6 +50,9 @@ private static ElasticsearchCluster createCluster() {
             .setting("xpack.security.enabled", "true")
             .keystore("bootstrap.password", "x-pack-test-password")
             .user("x_pack_rest_user", "x-pack-test-password");
+        if (initTestSeed().nextBoolean()) {
```
Comment (martijnvg) [inline modules/data-streams/src/yamlRestTest/java/org/elasticsearch/datastreams/DataStreamsClientYamlTestSuiteIT.java:53]: I think we just need to use `nextBoolean()` here?

File: test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java [hunk 2]
```diff
@@ -381,7 +380,7 @@ public void append(LogEvent event) {
         JAVA_ZONE_IDS = ZoneId.getAvailableZoneIds().stream().filter(unsupportedZoneIdsPredicate.negate()).sorted().toList();
     }
 
-    static Random initTestSeed() {
+    protected static Random initTestSeed() {
```
Comment (martijnvg) [inline test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java:383]: and then undo the changes to this file?

**Review by martijnvg — COMMENTED** [review]
Left two questions.

File: modules/data-streams/src/yamlRestTest/java/org/elasticsearch/datastreams/DataStreamsClientYamlTestSuiteIT.java (hunk 1 above)
Comment (kkrik-es) [reply modules/data-streams/src/yamlRestTest/java/org/elasticsearch/datastreams/DataStreamsClientYamlTestSuiteIT.java:53]: It's not possible to use this function here since the executing thread doesn't contain a `RandomizedContext`. This is a workaround for using the random seed in early initialization logic.

Comment (martijnvg) [reply modules/data-streams/src/yamlRestTest/java/org/elasticsearch/datastreams/DataStreamsClientYamlTestSuiteIT.java:53]: gotcha, yes that is tricky.

**Review by martijnvg — APPROVED** [review]
LGTM

Serverless failure looks unrelated to this change.
